### ISSUE TITLE
Consider using kcptun for transport

### ISSUE THREAD

**Issue by thesyncim** [schollz/croc#46]
just an idea
Not sure if it's worth it

**Comment by schollz** [comment]
Its cool, I was looking at it today and wondering about it. I just tried it, using a single TCP socket.

I tried using `-mode fast3 -crypt none -nocomp` which should be fastest I assume?  Pipelines was:

```
croc client -> kcptun client on localhost -> kcptun server on cowyo.com ->  croc relay -> croc client
```

The transfer to the server via `kcptun` is fast! Like 3-4 times faster. However the overall transfer is not any different, likely because its still limited? Or is it only faster because `kcptun` is holding it in memory while it waits to be read from the server? I'm not really sure I understand how it works.

A big snag: for `kcptun` to be used on both sides of the transfer, the receiver will have act as a server as well. I want to avoid this because it requires port-forwarding (unless there is a way that doesn't need port forwarding?).
